pkg/plugin/validator.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package plugin provides the implementation of the JVS plugin interface.
+package plugin
+
+import (
+	"context"
+	"crypto/rsa"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/abcxyz/pkg/githubapp"
+	"github.com/google/go-github/v55/github"
+	"github.com/lestrrat-go/jwx/v2/jwk"
+)
+
+type Validator struct {
+	cfg        *PluginConfig
+	decodedPem *rsa.PrivateKey
+	client     *github.Client
+
+	// for testing only
+	fakeGithubIssue *github.Issue
+}
+
+// ExchangeResponse is the GitHub API response of requesting an access token
+// for the GitHub App installation with requested repositories and permissions.
+type ExchangeResponse struct {
+	AccessToken string `json:"token"`
+}
+
+// pluginGithubIssue contains the required attribute parsed from
+// the issue URL.
+type pluginGithubIssue struct {
+	Owner       string
+	RepoName    string
+	IssueNumber int
+}
+
+type ValidatorOption func(*Validator)
+
+func WithFakeGithubIssue(i *github.Issue) ValidatorOption {
```

Comment:
We rarely provide a explicit and dedicated fake option. Why not directly set `fakeGitHubIssue` in test?

Reply:
Changed to directly set `fakeGitubIssue`